src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/CpsPackageReferenceProject.cs

```diff
@@ -30,11 +30,13 @@
 namespace NuGet.PackageManagement.VisualStudio
 {
     /// <summary>
-    /// Represents a project object associated with new VS "15" CPS project with package references.
+    /// Represents a project object associated with Common Project System (CPS) project that has opt'd
+    /// into package references. This includes, but may not be limited to, .NET Project System,
+    /// C++ (with PackageReference support) and MSIX deployment projects.
     /// Key feature/difference is the project restore info is pushed by nomination API and stored in 
     /// a cache. Factory method retrieving the info from the cache should be provided.
     /// </summary>
-    public class NetCorePackageReferenceProject : BuildIntegratedNuGetProject
+    public class CpsPackageReferenceProject : BuildIntegratedNuGetProject
```

Comment:
Are the warning from the assets file going to be handled by the project-system?
This type assumes that.

Reply:
There's two things that make work;

1) The target that reads the errors and produces it during the build (this is a .NET SDK thing)
2) The project system that pushes errors/warnings produced during a design-time build to the error list so this appears to happen without a build is CPS. You can assume that we need to move 1) to the correct layer for all project types using 2) work the same way.

Reply:
I think 1. might be challenging cause of the whole *who ships NuGet dlls* thing. 

The SDK carries it's own NuGet assemblies. Other components don't compile against NuGet (which is why Drew moved the solution explorer expansion in our codebase). 

Is there a component that's shared among SDK + C++ that knows about the assets file?

Reply:
We always wanted NuGet to own the intepretation of the assets file, a quite a few managers of NuGet ago pushed back on it.

C++ will be always including .NET SDK.
MSIX, we'll need to move stuff around.